pkg/apis/kops/validation/legacy.go

```diff
@@ -135,6 +133,16 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 			if err != nil {
 				allErrs = append(allErrs, field.Invalid(fieldSpec.Child("networkCIDR"), c.Spec.NetworkCIDR, "Cluster had an invalid networkCIDR"))
 			}
+			if kops.CloudProviderID(c.Spec.CloudProvider) == kops.CloudProviderDO {
+				// verify if the NetworkCIDR is in a private range as per RFC1918
+				if !networkCIDR.IP.IsPrivate() {
+					allErrs = append(allErrs, field.Invalid(fieldSpec.Child("networkCIDR"), c.Spec.NetworkCIDR, "Cluster had a networkCIDR outside the private IP range"))
+				}
+				// verify if networkID is not specified. In case of DO, this is mutually exclusive.
+				if c.Spec.NetworkID != "" {
+					allErrs = append(allErrs, field.Invalid(fieldSpec.Child("networkCIDR"), c.Spec.NetworkCIDR, "DO doesn't support specifying both NetworkID and NetworkCIDR together"))
```

Comment:
nit: we tend to use `field.Forbidden` when the value is legal, but not allowed in combination with other options.

Reply:
Sure, updated.